src/unix/newlib/mod.rs

```diff
@@ -28,9 +28,16 @@ pub type socklen_t = u32;
 pub type speed_t = u32;
 pub type suseconds_t = i32;
 pub type tcflag_t = ::c_uint;
-pub type time_t = i32;
 pub type useconds_t = u32;
 
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
```

Comment:
Is this actually the right condition to use in `cfg`? I was just following the other examples I had seen, but might this cause problems in the future for other `horizon` targets (notably, I think, Switch?)

Reply:
This is how the std code currently works, so it's fine for now. If the Switch is added it could be further filtered.